### PR TITLE
One more test for concrete set unification

### PR DESCRIPTION
**Pull request by dfilaretti**
EDIT: the `z3` error (mentioned below) were due to using an older version. After building the latest version of `z3` from source I am not getting that error anymore. 

The output I see is still not what I think should be the expected result, so I kept this test into the `in-progress` folder. 

--- OLDER ---

This is the error I used to get before upgrading `z3` to latest version: 

```
*** Data.SBV: Unexpected non-success response from Z3:
***
***    Sent      : (set-option :global-declarations true)
***    Expected  : success
***    Received  : (error "line 2 column 33: unknown parameter 'global_declarations'
***                Legal parameters are:
***                  auto_config (bool) (default: true)
***                  debug_ref_count (bool) (default: false)
***                  dump_models (bool) (default: false)
***                  memory_high_watermark (unsigned int) (default: 0)
***                  memory_max_alloc_count (unsigned int) (default: 0)
***                  memory_max_size (unsigned int) (default: 0)
***                  model (bool) (default: true)
***                  model_validate (bool) (default: false)
***                  proof (bool) (default: false)
***                  rlimit (unsigned int) (default: 4294967295)
***                  smtlib2_compliant (bool) (default: false)
***                  timeout (unsigned int) (default: 4294967295)
***                  trace (bool) (default: false)
***                  trace_file_name (string) (default: z3.log)
***                  type_check (bool) (default: true)
***                  unsat_core (bool) (default: false)
***                  verbose (unsigned int) (default: 0)
***                  warning (bool) (default: true)
***                  well_sorted_check (bool) (default: false)")
***
***    Exit code : ExitFailure (-15)
***    Executable: /usr/bin/z3
***    Options   : -nw -in -smt2
***
***    Reason    : Backend solver reports it does not support this option.
***                Check the spelling, and if correct please report this as a
***                bug/feature request with the solver!

[Error] Critical: Haskell backend returned non-zero exit code
```

###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`
- [ ] Style conformance: `stylish-haskell`

---

### CODE REVIEW

**Comment by u--** [comment]
Strange. The SMT interface is supposed to ignore anything it can't translate. I'll take a look.

**Comment by dfilaretti** [comment]
After updating `z3` to  4.8.2 (built from source) I am not getting that error anymore (I was using an older version from Ubuntu package manager)

**Comment by dfilaretti** [comment]
But I am still not getting the result I'd expect...

File: src/main/k/in-progress/tests/collections/set/set-unify-concrete-2/Makefile [hunk 1]
```diff
@@ -0,0 +1,34 @@
+TOP != git rev-parse --show-toplevel
+include $(TOP)/include.mk
+
+KOMPILED := set-unify-concrete-2-kompiled
+DEFINITION := $(KOMPILED)/definition.kore
+
+$(DEFINITION): set-unify-concrete-2.k $(KOMPILE_TARGETS)
```
Comment (ttuegel) [inline src/main/k/in-progress/tests/collections/set/set-unify-concrete-2/Makefile]:
```suggestion
$(DEFINITION): set-unify-concrete-2.k $(K)
```

File: src/main/k/in-progress/tests/collections/set/set-unify-concrete-2/Makefile [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	$(KOMPILE) $(KOMPILE_OPTS) $< --syntax-module set-unify-concrete-2
+
+%.set-unify-concrete-2.kore: %.set-unify-concrete-2 $(DEFINITION) $(KRUN_TARGETS)
```
Comment (ttuegel) [inline src/main/k/in-progress/tests/collections/set/set-unify-concrete-2/Makefile]:
```suggestion
%.set-unify-concrete-2.kore: %.set-unify-concrete-2 $(DEFINITION) $(K) $(KORE_EXEC)
```

File: src/main/k/in-progress/tests/collections/set/set-unify-concrete-2/Makefile [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+	$(KRUN) $(KRUN_OPTS) $< --dry-run
+
+%.output: %.set-unify-concrete-2 $(DEFINITION) $(KRUN_TARGETS)
```
Comment (ttuegel) [inline src/main/k/in-progress/tests/collections/set/set-unify-concrete-2/Makefile]:
```suggestion
%.output: %.set-unify-concrete-2 $(DEFINITION) $(K) $(KORE_EXEC)
```

File: src/main/k/in-progress/tests/collections/set/set-unify-concrete-2/Makefile [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+	$(KRUN) $(KRUN_OPTS) $< --output-file $@
+
+%.krun: %.set-unify-concrete-2 $(DEFINITION) $(KRUN_TARGETS)
```
Comment (ttuegel) [inline src/main/k/in-progress/tests/collections/set/set-unify-concrete-2/Makefile]:
```suggestion
%.krun: %.set-unify-concrete-2 $(DEFINITION) $(K) $(KORE_EXEC)
```

**Pull request merged**